core/src/components/tab-button/tab-button.ionic.scss

```diff
@@ -86,3 +90,75 @@
 :host(.tab-button-shape-rectangular) {
   @include globals.border-radius(globals.$ion-border-radius-0);
 }
+
+// Tab Button: Badge
+// --------------------------------------------------
+
+:host ::slotted(ion-badge.badge-xxsmall),
+:host ::slotted(ion-badge.badge-xsmall),
+:host ::slotted(ion-badge.badge-small),
+:host ::slotted(ion-badge.badge-medium),
+:host ::slotted(ion-badge.badge-large),
+:host ::slotted(ion-badge.badge-xlarge) {
+  min-width: ion-vars.$tab-ionic-button-badge-size;
+  height: ion-vars.$tab-ionic-button-badge-size;
+
+  font-size: ion-vars.$tab-ionic-button-badge-font-size;
+
+  line-height: ion-vars.$tab-ionic-button-badge-line-height;
+}
+
+:host ::slotted(ion-badge:empty) {
+  @include mixins.padding(0, 0);
+
+  min-width: ion-vars.$tab-ionic-button-badge-size-emtpy;
+  height: ion-vars.$tab-ionic-button-badge-size-emtpy;
```

Comment:
This isn't needed. I was able to remove it on my end and verified that the sizes are still correct when using the correct `size="small"` prop. I made a comment in `badge/test/hint/index.html` that explains it.

Reply:
Strange, on my end will get the default badge size based on the given size... Which makes sense since empty is not being set anywhere else...

With min-width and height properties:
<img width="75" alt="Screenshot 2025-03-19 at 17 04 41" src="https://github.com/user-attachments/assets/22103a5e-399a-4055-94f3-5b8b3a197876" />


Without:
<img width="84" alt="Screenshot 2025-03-19 at 17 00 53" src="https://github.com/user-attachments/assets/de47529b-b949-44af-a445-5d8e3842f4bd" />

Reply:
Kept them since they're necessary.

Reply:
Update: Same as [here](https://github.com/ionic-team/ionic-framework/pull/30246#discussion_r2003913865)